ppr-api/src/ppr_api/utils/validators/financing_validator.py

```diff
@@ -0,0 +1,198 @@
+            'vehicleCollateral' in json_data and json_data['vehicleCollateral']:
+        error_msg = VC_NOT_ALLOWED
+    elif reg_type in VC_REQUIRED_LIST and ('vehicleCollateral' not in json_data or not json_data['vehicleCollateral']):
+        error_msg = VC_REQUIRED
+    elif reg_type in VC_MH_ONLY_LIST:
+        for collateral in json_data['vehicleCollateral']:
+            if 'type' in collateral and collateral['type'] != 'MH':
+                error_msg = VC_MH_ONLY
+    elif reg_type == model_utils.REG_TYPE_REPAIRER_LIEN:
+        for collateral in json_data['vehicleCollateral']:
+            if 'type' in collateral and collateral['type'] == 'MH':
+                error_msg = VC_MH_NOT_ALLOWED
+
+    return error_msg
+
+
+def validate_other_description(json_data, reg_type: str):
+    """Validate otherTypeDescription by registration type."""
+    if reg_type == model_utils.REG_TYPE_OTHER and \
+       ('otherTypeDescription' not in json_data or str(json_data['otherTypeDescription']).strip() == ''):
+        return OT_MISSING_DESCRIPTION
+    if reg_type != model_utils.REG_TYPE_OTHER and \
+       ('otherTypeDescription' in json_data and str(json_data['otherTypeDescription']).strip() != ''):
+        return OT_NOT_ALLOWED
+    return ''
+
+
+def validate_trust_indenture(json_data, reg_type: str):
+    """Validate trustIndenture by registration type."""
+    if reg_type != model_utils.REG_TYPE_SECURITY_AGREEMENT and 'trustIndenture' in json_data:
+        return TI_NOT_ALLOWED
+    return ''
+
+
+def validate_rl(json_data, reg_type: str):
+    """Validate Repairer's Lien."""
+    error_msg = ''
+    if reg_type != model_utils.REG_TYPE_REPAIRER_LIEN:
+        if 'lienAmount' in json_data and json_data['lienAmount']:
+            error_msg = LA_NOT_ALLOWED
+        if 'surrenderDate' in json_data and json_data['surrenderDate']:
+            error_msg += SD_NOT_ALLOWED
+        return error_msg
+
+    if 'lienAmount' not in json_data or str(json_data['lienAmount']).strip() == '':
+        error_msg = RL_AMOUNT_REQUIRED
+    if 'surrenderDate' not in json_data or str(json_data['surrenderDate']).strip() == '':
+        error_msg += RL_DATE_REQUIRED
+    else:
+        try:
+            surrender_date = model_utils.ts_from_date_iso_format(json_data['surrenderDate'])
+            if surrender_date:
+                test_date = model_utils.today_ts_offset(21, False)
+                if surrender_date.timestamp() < test_date.timestamp():
+                    error_msg += RL_DATE_INVALID
+        except ValueError:
+            error_msg += RL_DATE_INVALID
+
+    return error_msg
+
+
+def get_registration_class(reg_type: str):
+    """Derive the registration class from the registration type."""
+    try:
+        reg_class = model_utils.REG_TYPE_TO_REG_CLASS[reg_type]
+        if reg_class in (model_utils.REG_CLASS_CROWN, model_utils.REG_CLASS_MISC, model_utils.REG_CLASS_PPSA):
```

Comment:
```suggestion
        if (reg_class := model_utils.REG_TYPE_TO_REG_CLASS[reg_type]) in \
        (model_utils.REG_CLASS_CROWN, model_utils.REG_CLASS_MISC, model_utils.REG_CLASS_PPSA):
```
the walrus is your friend :=

Reply:
Suggested change made.